src/ipa_pc/mod.rs

```diff
@@ -1004,8 +954,7 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             batch_polynomial += (cur_challenge, &polynomial);
 
             // lambda^i
-            cur_challenge = opening_challenges(opening_challenge_counter);
-            opening_challenge_counter += 1;
+            cur_challenge = cur_challenge * &lambda;
```

Comment:
If we do not target recursion, then squeezing again would be fine, too.

Reply:
Let's keep it uniform and squeeze new challenges everywhere instead of squeeze one and computing the powers.
Let's put a comment like before, that this is not recursion friendly

Reply:
With squeezing this part doesn't work properly. Here we are using batching which is described in document with using the powers. Are you sure that here we can replace power with squeezing?

Reply:
If it is about security, we can sample "fresh" challenges for every polynomial in the batch. If one does not target recursive proofs, then hashing might be cheaper than doing a field multiplication (although I doubt that this is the case in practice).
For now, we need to decide for one of the two approaches. As we switched to powers in all other places, we can keep with the powers here, too.

Reply:
Ok, so no changes needed here for now?